packages/salesforcedx-vscode-core/src/orgBrowser/metadataCmp.ts

```diff
@@ -132,19 +159,36 @@ export class ComponentUtils {
     );
 
     let componentsList: string[];
+    const connection = await Connection.create({
+      authInfo: await AuthInfo.create({
+        username: await OrgAuthInfo.getUsername(defaultOrg)
+      })
+    });
     if (metadataType === 'CustomObject' && folder) {
-      componentsList = await this.buildCustomObjectFieldsList(
-        folder,
-        defaultOrg,
-        componentsPath
-      );
+      if (forceRefresh || !fs.existsSync(componentsPath)) {
```

Comment:
With the code from line 167 through 205, how about refactoring lines 168-183 into a function, and lines 185-205 into another function?

Reply:
Thank you for the suggestion. I have refactored it.